src/scaffoldmaker/utils/cylindermesh.py

```diff
@@ -0,0 +1,477 @@
+            for c in range(3)]
+        rcx.append(tmdx)
+        rcx.append(tmux)
+        rcd3 = [vector.setMagnitude(tmdd3, -1), vector.setMagnitude(tmdd3, -1)]
+        rscx, rscd1 = sampleCubicHermiteCurves(rcx, rcd3, self._shield.elementsCountUp, lengthFractionStart=1,
+                                               arcLengthDerivatives=True)[0:2]
+
+        # get d2, d3
+        rscd2 = []
+        rscd3 = []
+        for n in range(len(rscx)):
+            d3 = vector.normalise(
+                [btx[n3][self._elementsCountUp][self._elementsCountAcrossMinor][c] - btx[n3][self._elementsCountUp][0][c] for
+                 c in range(3)])
+            d2 = vector.normalise(vector.crossproduct3(d3, rscd1[n]))
+            rscd2.append(d2)
+            rscd3.append(d3)
+
+        return rscx, rscd1, rscd2, rscd3
+
+    def createRegularRowCurves(self, n3, rscx, rscd1, rscd3):
+        '''
+        generate curves along regular rows using the mirror curve obtained from createMirrorCurve.
+        :param n3: Index of along cylinder axis coordinates to use
+        '''
+        btx = self._shield.px
+        btd1 = self._shield.pd1
+        btd2 = self._shield.pd2
+        btd3 = self._shield.pd3
+
+        elementsCountRim = 0
+        for n2 in range(elementsCountRim + 2, self._elementsCountUp + 1):
+            btx[n3][n2], btd3[n3][n2], pe, pxi, psf = sampleCubicHermiteCurves(
+                [btx[n3][n2][0], rscx[n2], btx[n3][n2][-1]],
+                [vector.setMagnitude(btd3[n3][n2][0], -1.0), rscd3[n2], btd3[n3][n2][-1]], self._elementsCountAcrossMinor,
+                lengthFractionStart=1, lengthFractionEnd=1, arcLengthDerivatives=True)
+            btd1[n3][n2] = \
+            interpolateSampleCubicHermite([[-btd1[n3][n2][0][c] for c in range(3)], rscd1[n2], btd1[n3][n2][-1]],
+                                          [[0.0, 0.0, 0.0]] * 3, pe, pxi, psf)[0]
+            btd3[n3][n2][0] = [-btd3[n3][n2][0][c] for c in range(3)]
+            btd1[n3][n2][0] = [-btd1[n3][n2][0][c] for c in range(3)]
+
+    def createRegularColumnCurves(self, n3):
+        '''
+        up regular columns of shield: get d1, initial d3 below regular rows
+        '''
+        btx = self._shield.px
+        btd1 = self._shield.pd1
+        btd2 = self._shield.pd2
+        btd3 = self._shield.pd3
+
+        for n1 in range(2, self._elementsCountAcrossMinor - 1):
+            tx, td1, pe, pxi, psf = sampleCubicHermiteCurves(
+                [btx[n3][0][n1], btx[n3][2][n1]], [[-btd3[n3][0][n1][c] for c in range(3)], btd1[n3][2][n1]], 2,
+                lengthFractionStart=1, arcLengthDerivatives=True)  # GRC fudge factor rvSulcusEdgeFactor
+            for n2 in range(3, self._elementsCountUp + 1):
+                tx.append(btx[n3][n2][n1])
+                td1.append(btd1[n3][n2][n1])
+            td1 = smoothCubicHermiteDerivativesLine(tx, td1, fixStartDirection=True)
```

Comment:
If this is the code smoothing d1 up to the mirror line, need to ensure that the mirror line d1 has been set to be normal to the mirror line and use fixEndDirection=True here. This will maintain symmetry.